### PR TITLE
autofix for PT003

### PR DESCRIPTION
**Pull request by sbrugman**
Closes https://github.com/charliermarsh/ruff/issues/2260

### CODE REVIEW

File: src/rules/flake8_pytest_style/rules/fixture.rs [hunk 1]
```diff
@@ -132,10 +132,17 @@ fn check_fixture_decorator(checker: &mut Checker, func_name: &str, decorator: &E
...
                             violations::ExtraneousScopeFunction,
                             Range::from_located(scope_keyword),
-                        ));
+                        );
+                        if checker.patch(diagnostic.kind.rule()) {
+                            diagnostic.amend(Fix::deletion(
+                                scope_keyword.location,
+                                scope_keyword.end_location.unwrap(),
+                            ));
+                        }
```
Comment (charliermarsh) [inline src/rules/flake8_pytest_style/rules/fixture.rs:144]: I think we need to be a bit more careful here. E.g. with `@pytest.fixture(scope="function",)`, would this leave the trailing comma and thus lead to a syntax error? What if this is the last argument, and we don't remove the preceding comma?

Trying to find an example of where we handle this... I guess `remove_class_def_base`? But maybe not the best implementation.

**Comment by edgarrmondragon** [comment]
[`pytest.fixture`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.fixture) can have more arguments, this seems to preserve them but what happens to the comma?

```python
@pytest_fixture(scope='function', name='my_fixture')
def dummy():
    return 1
```

**Comment by sbrugman** [comment]
Thanks for the pointers @edgarrmondragon and @charliermarsh.
I've changed my naive solution to take the `keywords.iter()` and check for a next value, if present it searches for the comma and extended the tests.
(Refactored the related violations with it)

File: resources/test/fixtures/flake8_pytest_style/PT003.py [hunk 2]
```diff
@@ -14,3 +14,16 @@ def ok_other_scope():
...
+    ...
+
+
+@pytest.fixture(name="my_fixture", scope="function", **kwargs)
+def error_second_arg():
+    ...
+
+@pytest.fixture("my_fixture", scope="function", **kwargs)
+def error_arg():
+    ...
```
Comment (charliermarsh) [inline resources/test/fixtures/flake8_pytest_style/PT003.py:29]: I think there are still a few cases that aren't quite handled correctly:

```py
@pytest.fixture(
    scope="function",
    name="my_fixture",
)
def error_multiple_args():
    ...


@pytest.fixture(name="my_fixture", scope="function")
def error_multiple_args():
    ...
```

These give:

```py
@pytest.fixture(
    
    name="my_fixture",
)
def error_multiple_args():
    ...


@pytest.fixture(name="my_fixture", )
def error_multiple_args():
    ...
```

Sorry for all the churn, but do you mind tweaking to address those?

Comment (sbrugman) [reply resources/test/fixtures/flake8_pytest_style/PT003.py:29]: Will do, should have tested a bit better

Comment (charliermarsh) [reply resources/test/fixtures/flake8_pytest_style/PT003.py:29]: Na don't sweat it. I only know to try these because I've run into them before :joy:

Comment (sbrugman) [reply resources/test/fixtures/flake8_pytest_style/PT003.py:29]: Would be nice to turn them into utilities after (e.g. `remove_argument`)

Comment (charliermarsh) [reply resources/test/fixtures/flake8_pytest_style/PT003.py:29]: Strongly agree!

**Review by charliermarsh — CHANGES_REQUESTED** [review]
(Just requesting changes to help manage my own queue, no rush of course.)

**Comment by sbrugman** [comment]
No problem, working on a general fix helper for removing (keyword)args. Will try to complete today.

**Comment by charliermarsh** [comment]
Genuinely no rush, take your time :)

**Comment by sbrugman** [comment]
@charliermarsh Introduced a helper function in `autofix/helpers.rs`. 
This is now used here, and for pyupgrade's `object` base removal (class definition and function call use the same `arguments` part in the python [grammar](https://docs.python.org/3/reference/grammar.html)).

(I also see opportunity to use it to autofix `PD002`)

**Pull request merged**